### PR TITLE
Hide passwords in URLs on the /config endpoint.

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom@grafana.com>

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by Wing924** [comment]
I think we also need to mask db and cassandra password.

**Comment by Wing924** [comment]
How about create `flagext.Password` for those secret?

**Comment by pracucci** [comment]
> How about create flagext.Password for those secret?

Makes sense. Maybe `flagext.Secret` is more appropriate, cause it's not just passwords but sensitive data in general.

**Comment by tomwilkie** [comment]
> S3 secret access key, GCS service_account

AFAICT these are part of a URL, so handled as part of the original `URLValue` changes.

> Possibly others, I haven't fully investigated the whole config

Also Cassandra, didn't find anymore.

**Comment by pracucci** [comment]
> AFAICT these are part of a URL, so handled as part of the original URLValue changes.

@tomwilkie We're both right. You're right about the chunks storage, while I had the blocks storage in mind. Please take a look here:

https://github.com/cortexproject/cortex/blob/9cec079a434079e3989f924e323c99231df226a2/pkg/storage/tsdb/backend/s3/config.go#L8-L14

https://github.com/cortexproject/cortex/blob/9cec079a434079e3989f924e323c99231df226a2/pkg/storage/tsdb/backend/gcs/config.go#L8-L11

**Comment by tomwilkie** [comment]
> cortex/pkg/storage/tsdb/backend/gcs/config.go

I didn't think ServiceAccount is secret, is it?  I thought GCS credentials were always read from the env...

File: pkg/util/flagext/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestSecretdYAML(t *testing.T) {
+	// Test embedding of Password.
```
Comment (pracucci) [inline pkg/util/flagext/secret_test.go]: Please replace `Password` with `Secret` in the comments, for correctness.

File: pkg/util/flagext/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+		var actualStruct TestStruct
+		yamlSecret := []byte(`secret: pa55w0rd
+`)
+		err = yaml.Unmarshal(yamlSecret, &actualStruct)
+		require.NoError(t, err)
+		assert.Equal(t, testStruct, actualStruct)
+	}
+
+	// Test no url set in Password.
```
Comment (pracucci) [inline pkg/util/flagext/secret_test.go]: > no url set

I guess you mean "no value set"

**Comment by tomwilkie** [comment]
Thank you for your review @pracucci!

**Pull request merged**